File: Backend/BloodDonationSupport/BloodDonationSupport.Infrastructure/Persistence/Models/Address.cs
namespace BloodDonationSupport.Infrastructure.Persistence.Models;

public partial class Address
{
    public long AddressId { get; set; }

    public string? Line1 { get; set; }

    public string? District { get; set; }

    public string? City { get; set; }

    public string? Province { get; set; }

    public string? Country { get; set; }

    public string? PostalCode { get; set; }

    public string? NormalizedAddress { get; set; }

    public string? PlaceId { get; set; }

    public decimal? ConfidenceScore { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual ICollection<Donor> Donors { get; set; } = new List<Donor>();

    public virtual ICollection<HealthcareFacility> HealthcareFacilities { get; set; } = new List<HealthcareFacility>();

    public virtual ICollection<Request> Requests { get; set; } = new List<Request>();
}

File: Frontend/src/types/address.ts
export interface Address {
  id?: number;
  line1: string;
  district?: string;
  city: string;
  province: string;
  country: string;
  postalCode?: string;
  normalizedAddress?: string;
  placeId?: string;
  confidenceScore?: number;
  latitude?: number;
  longitude?: number;
}

export interface AddressResponse {
  success: boolean;
  message: string | null;
  data: Address;
}

export interface CreateAddressRequest {
  line1: string;
  district?: string;
  city: string;
  province: string;
  country: string;
  postalCode?: string;
  normalizedAddress?: string;
  placeId?: string;
  confidenceScore?: number;
  latitude?: number;
  longitude?: number;
}

export interface UpdateAddressRequest extends Partial<CreateAddressRequest> {
  id?: number;
}

